S2 Master Fails to build with clang14

There are numerous errors when building with g++12.3.0 in Ubuntu 22.04 off master [035272b63b313d2c79c7e8b9606f022b0570fe7b]

The following errors repeat a bunch of times.  Building using g++11.4 works fine.  
```
2-src/src/s2/base/port.h:35:3: error: no matching function for call to 'operator delete'
  ::operator delete(ptr, size);
  ^~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/new:180:13: note: candidate function not viable: no known conversion from 'size_t' (aka 'unsigned long') to 'void *' for 2nd argument; take the address of the argument with &
inline void operator delete  (void*, void*) _GLIBCXX_USE_NOEXCEPT { }
            ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/new:144:6: note: candidate function not viable: no known conversion from 'size_t' (aka 'unsigned long') to 'const std::nothrow_t' for 2nd argument
void operator delete(void*, const std::nothrow_t&) _GLIBCXX_USE_NOEXCEPT
     ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/new:153:6: note: candidate function not viable: no known conversion from 'size_t' (aka 'unsigned long') to 'std::align_val_t' for 2nd argument
void operator delete(void*, std::align_val_t)
     ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/new:130:6: note: candidate function not viable: requires 1 argument, but 2 were provided
void operator delete(void*) _GLIBCXX_USE_NOEXCEPT
     ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/new:155:6: note: candidate function not viable: requires 3 arguments, but 2 were provided
void operator delete(void*, std::align_val_t, const std::nothrow_t&)
     ^
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,6 +92,12 @@ else()
     # match [-Wattributes]
     add_definitions(-Wno-attributes)
     add_definitions(-Wno-deprecated-declarations)
+    # `base/port.h` uses sized deallocation, which should be enabled by
+    # default for C++14 and later.  There appears to be a bug with clang
+    # < 19 causing this not to be enabled.
+    # https://github.com/google/s2geometry/issues/411#issuecomment-2726949607
+    # This can be removed when clang19 is the minimum supported version.
+    add_compile_options(-fsized-deallocation)
 endif()
 
 # If OpenSSL is installed in a non-standard location, configure with

